crates/anvil/src/eth/backend/mem/storage.rs

```diff
@@ -415,13 +418,44 @@ impl MinedTransaction {
         })
     }
 
-    pub fn geth_trace(&self, opts: GethDefaultTracingOptions) -> DefaultFrame {
-        GethTraceBuilder::new(self.info.traces.clone(), TracingInspectorConfig::default_geth())
-            .geth_traces(
-                self.receipt.gas_used().to::<u64>(),
-                self.info.out.clone().unwrap_or_default().0.into(),
-                opts,
-            )
+    pub fn geth_trace(&self, opts: GethDebugTracingOptions) -> GethTrace {
+        let GethDebugTracingOptions { config, tracer, tracer_config, .. } = opts;
+
+        if let Some(tracer) = tracer {
+            return match tracer {
+                GethDebugTracerType::BuiltInTracer(tracer) => match tracer {
+                    GethDebugBuiltInTracerType::FourByteTracer => {
+                        let inspector = FourByteInspector::default();
+                        return FourByteFrame::from(inspector).into()
+                    }
+                    GethDebugBuiltInTracerType::CallTracer => {
+                        let call_config = tracer_config.into_call_config().unwrap();
+
+                        return GethTraceBuilder::new(
+                            self.info.traces.clone(),
+                            TracingInspectorConfig::from_geth_config(&config),
+                        )
+                        .geth_call_traces(call_config, self.receipt.gas_used())
```

Comment:
I think this needs to be converted:

.geth_call_traces(call_config, self.receipt.gas_used().to::<u64>())